eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -3,93 +3,379 @@
+  if ($DeleteAfter) {
+    $deleteDate = $deleteAfter -as [DateTime]
+    return $deleteDate -and [datetime]::UtcNow -gt $deleteDate
+  }
+  return $false
+}
+
+function HasException([object]$ResourceGroup) {
+  if ($Exceptions.Count -and $Exceptions.Contains($ResourceGroup.ResourceGroupName)) {
+    Write-Host " Skipping allowed resource group '$($ResourceGroup.ResourceGroupName)' because it is in the allow list '$AllowListPath'"
+    return $true
+  }
+  return $false
+}
 
-    if ($purgeableResourcesFromRG) {
-      $purgeableResources += $purgeableResourcesFromRG
-      Write-Verbose "Found $($purgeableResourcesFromRG.Count) potentially purgeable resources in resource group $($rg.ResourceGroupName)"
+function FindOrCreateDeleteAfterTag {
+  [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+  param(
+    [object]$ResourceGroup
+  )
+
+  $deleteAfter = GetTag $ResourceGroup "DeleteAfter"
+  if (!$deleteAfter -or !($deleteAfter -as [datetime])) {
+    $deleteAfter = [datetime]::UtcNow.AddHours($DeleteAfterHours)
+    if ($Force -or $PSCmdlet.ShouldProcess("$($ResourceGroup.ResourceGroupName) [DeleteAfter (UTC): $deleteAfter]", "Adding DeleteAfter Tag to Group")) {
+      Write-Host "Adding DeleteAfer tag with value '$deleteAfter' to group '$($ResourceGroup.ResourceGroupName)'"
+      $ResourceGroup | Update-AzTag -Operation Merge -Tag @{ DeleteAfter = $deleteAfter }
     }
-    Write-Verbose "Deleting group: $($rg.ResourceGroupName)"
-    Write-Verbose "  tags $($rg.Tags | ConvertTo-Json -Compress)"
-    Write-Host ($rg | Remove-AzResourceGroup -Force -AsJob).Name
   }
 }
 
-# Purge all the purgeable resources and get a list of resources (as a collection) we need to follow-up on.
-Write-Host "Attempting to purge $($purgeableResources.Count) resources."
-$failedResources = @(Remove-PurgeableResources $purgeableResources -PassThru)
-if ($failedResources) {
-  Write-Warning "Timed out deleting the following $($failedResources.Count) resources. Please file an IcM ticket per resource type."
-  $failedResources | Sort-Object AzsdkResourceType, AzsdkName | Format-Table -Property @{l='Type'; e={$_.AzsdkResourceType}}, @{l='Name'; e={$_.AzsdkName}}
+function HasDeleteLock() {
+  return $false
+}
+
+function DeleteOrUpdateResourceGroups() {
+  [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+  param()
+
+  if ($IsProvisionerApp) {
+    AddGithubUsersToAliasCache
+  }
+
+  Write-Verbose "Fetching groups"
+  $allGroups = @(Get-AzResourceGroup)
+  $toDelete = @()
+  $toUpdate = @()
+  Write-Host "Total Resource Groups: $($allGroups.Count)"
+
+  foreach ($rg in $allGroups) {
+    if (HasException $rg) {
+      continue
+    }
+    $deleteAfter = GetDeleteAfterTag $rg
+    if ($deleteAfter) {
+      if (HasExpiredDeleteAfterTag $deleteAfter) {
+        $toDelete += $rg
+      }
+      continue
+    }
+    # TODO: Remove $true and follow non-compliant group deletion
```

Comment:
Might be worth calling out why you are doing this and even add a link to an follow-up issue.